### DISCUSSION TITLE
Custom values for the player

### DISCUSSION THREAD

**Discussion by Timofffee** [Minestom/Minestom#1673] [Help] [answered]
Hi. I need to implement custom values that belong to the player. This could be the mana value. But I don't want to use the values of experience, food or air for this. How can I do this?

I looked at the option with Attributes, but it seems they don't suit me. I also looked at the Tags, but they also most likely do not suit me due to the fact that it is essentially a wrapper for NBT.

I looked at how health is stored and it is right in the player's class. This is good, but working with events, I can't just expand the player class. Or am I confusing something?

The current option, which seems to me the most preferable, is to create a class/record in which there will be a list of players who will have values that I do not need.

**Comment by GoldenStack** [comment, marked as answer]
You can actually extend the player class (I've done it before in a personal project and it works fine) but make sure to set the player provider so it actually gets used:
https://javadoc.minestom.net/net/minestom/server/network/ConnectionManager.html#setPlayerProvider(net.minestom.server.network.PlayerProvider)

Also make sure that, if you use fake players in addition to this, you don't have any bugs due to them not extending your special class.
